controllers/gatekeeper_controller.go

```diff
@@ -188,7 +205,38 @@ func (r *GatekeeperReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *GatekeeperReconciler) deployGatekeeperResources(gatekeeper *operatorv1alpha1.Gatekeeper) error {
+func (r *GatekeeperReconciler) deployGatekeeperResources(gatekeeper *operatorv1alpha1.Gatekeeper, platformName string) error {
+	if platformName == "OpenShift" {
+		for _, asset := range openshiftAssets {
```

Comment:
Let's try to re-use the original for loop below by just checking if we've hit the role asset, then check if `isOpenShift` and prefix the necessary directory.